truthy-falsy.js

```diff
@@ -0,0 +1,219 @@
+}
+
+// 8
+function isValidPassword(password) {
+  const specialCharacters = /[`!@#$%^&*()_\-+=\[\]{};':"\\|,.<>\/?~ ]/;
+  const passwordLengthCheck = password.trim().length >= 8;
+  const passwordCapitalCharactersCheck = /[A-Z]/.test(password) >= 1;
+  const passwordSpecialCharactersCheck = specialCharacters.test(password) >= 1;
+
+  if (passwordLengthCheck && passwordCapitalCharactersCheck && passwordSpecialCharactersCheck) {
+    return true;
+  }
+  return false;
+}
+console.log(isValidPassword('passSword!'));
+
+// 9
+function isPasswordLengthValid(password) {
+  if (password.trim().length >= 8) {
+    return 1;
+  }
+  return 0;
+}
+console.log(isPasswordLengthValid('pass'));
+
+function hasUppercaseLetter(password) {
+  if (/[a-z]/.test(password) >= 1 ) {
+    return 1;
+  }
+  return 0;
+}
+
+function hasLowercaseLetter(password) {
+  if (/[A-Z]/.test(password) >= 1 ) {
+    return 1;
+  } return 0;
+}
+
+function hasSpecialLetter(password) {
+  const specialChars =/[`!@#$%^&*()_\-+=\[\]{};':"\\|,.<>\/?~ ]/;
+  if (specialChars.test(password) >= 1 ) {
+    return 1;
+  }
+  return 0;
+}
+
+function getPasswordRating(password) {
+  const passwordLength = isPasswordLengthValid(password);
+  const passwordLowercase= hasUppercaseLetter(password);
+  const passwordUppercase = hasLowercaseLetter(password);
+  const passwordSpecialLetter = hasSpecialLetter(password);
+  return passwordLength + passwordLowercase + passwordUppercase + passwordSpecialLetter;
+}
+console.log(getPasswordRating('passWord'));
+
+// 10
+
+function getVipDiscount(isVipMember) {
+  if (isVipMember) {
+    return 0.05;
+  }
+  return 1;
+}
+
+function getLoyaltyDiscount(loyaltyPoints) {
+  if (loyaltyPoints) {
+    return loyaltyPoints * 0.01;
+  }
+  return 0;
+}
+
+function getInternationalShippingPrice(isShippedInternationally) {
+  if (isShippedInternationally) {
+    return 10;
+  }
+  return 5;
+}
+
+function getTotalOrderCost (baseItemCost, isVipMember, loyaltyPoints, isShippedInternationally) {
+  const vipMemberDiscount = getVipDiscount(isVipMember);
+  const loyaltyDiscount = getLoyaltyDiscount(loyaltyPoints);
+  const shippingPrice = getInternationalShippingPrice(isShippedInternationally);
+
+  return baseItemCost - (baseItemCost * vipMemberDiscount) - loyaltyDiscount + shippingPrice
+}
+console.log(getTotalOrderCost(10,true,10,true))
+
+// 11
+function getDiscountDueToDaysLeftToShow(daysLeftToShow) {
+  if (daysLeftToShow > 30) {
+    return 0.10;
+  }
+  return 0;
+}
+
+function getShowOnWeekendDiscount(isOnWeekend) {
+  if (isOnWeekend) {
+    return 15;
+  }
+  return 0;
+}
+
+function getTicketPrice (baseTicketPrice, daysLeftToShow, isOnWeekend) {
+  const daysLeftToShowDiscount = getDiscountDueToDaysLeftToShow(daysLeftToShow);
+  const weekendDiscount = getShowOnWeekendDiscount(isOnWeekend);
+
+  return baseTicketPrice - (baseTicketPrice * daysLeftToShowDiscount) + weekendDiscount;
+}
+
+console.log(getTicketPrice(10, 31, true))
```

Comment:
You missed a semicolon here. Please try not to make the same mistakes multiple times